fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedWrappedDirectory.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * WrappedDirectory.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.codec;
+
+import com.apple.foundationdb.record.lucene.directory.FDBDirectory;
+import org.apache.lucene.store.BufferedChecksumIndexInput;
+import org.apache.lucene.store.ChecksumIndexInput;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FilterDirectory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.store.Lock;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static com.apple.foundationdb.record.lucene.directory.FDBDirectory.isEntriesFile;
+import static com.apple.foundationdb.record.lucene.directory.FDBDirectory.isSegmentInfo;
+
+public class LuceneOptimizedWrappedDirectory extends Directory {
+    private final FDBDirectory fdbDirectory;
+    private final Directory wrappedDirectory;
+
+    public LuceneOptimizedWrappedDirectory(Directory directory) {
+        this.wrappedDirectory = directory;
+        this.fdbDirectory = (FDBDirectory)FilterDirectory.unwrap(directory);
+    }
+
+    @Override
+    public String[] listAll() throws IOException {
+        throw new UnsupportedOperationException("Not Supported");
```

Comment:
Sonarqube warned ```Define a constant instead of duplicating this literal "Not Supported" 9 times.```. But I am not sure we want to fix it in this way.

Reply:
Eh, this is pretty minor, so it's probably fine as is. If we wanted to fix the warning, we could make a helper method in this class that returned an `UnsupportedOperationException` with the message pre-set just to avoid code dupe, but it's minor and so is probably fine.

Reply:
@tian-yizuo Can we just fix this so the warning is not there?  As an engineer I always curse at linters not happy with my code and then curse at projects not using linters.  Is this possible?